tests/dtm/test_multi_m0d_process_restart.py

```diff
@@ -303,3 +303,345 @@ def test_delete_after_multi_m0d_restart(self):
 
         self.test_completed = True
         self.log.info("ENDED: Verify delete after multiple m0d restart using pkill")
+
+    @pytest.mark.lc
+    @pytest.mark.dtm
+    @pytest.mark.tags("TEST-47158")
+    def test_continuous_write_read_during_k_m0d_restart_overlapping(self):
+        """
+        Verify Continuous Write/Reads during K process restart.
+        (Different m0d processes to be selected one after another with delay of 2 mins till k,
+        start each process one by one after 5 mins )
+        """
+        self.log.info("STARTED: Verify continuous WRITE and READ during different m0d restart "
+                      " using pkill")
+        log_file_prefix = 'test-47158'
+        que_1 = multiprocessing.Queue()
+        que_2 = multiprocessing.Queue()
+        m0d_que = multiprocessing.Queue()
+        bucket_list = []
+
+        self.log.info("Step 1: Create %s buckets for write operation during m0d restart",
+                      self.test_cfg['loop_count'])
+        for each in range(self.test_cfg['loop_count']):
+            bucket = f'{self.bucket_name}-{each}'
+            self.s3_test_obj.create_bucket(bucket)
+            bucket_list.append(bucket)
+
+        self.log.info("Step 2: Start write Operation:")
+        self.dtm_obj.perform_write_op(bucket_prefix=self.bucket_name,
+                                      object_prefix=self.object_name,
+                                      no_of_clients=self.test_cfg['clients'],
+                                      no_of_samples=self.test_cfg['samples'],
+                                      obj_size=self.test_cfg['size'],
+                                      log_file_prefix=log_file_prefix, queue=que_1)
+        resp = que_1.get()
+        assert_utils.assert_true(resp[0], resp[1])
+        workload_info_write = resp[1]
+
+        self.log.info("Step 3a: Start READ Operations in loop in background for data written in "
+                      "step 1:")
+        proc_read_op = multiprocessing.Process(target=self.dtm_obj.perform_ops,
+                                               args=(workload_info_write, que_1, False, True,
+                                                     True, self.test_cfg['loop_count']))
+        proc_read_op.start()
+        self.log.info("Step 3b: Start write Operations in loop in background:")
+        args = {'bucket_prefix': self.bucket_name, 'object_prefix': self.object_name,
+                'no_of_clients': self.test_cfg['clients'],
+                'no_of_samples': self.test_cfg['samples'], 'log_file_prefix': log_file_prefix,
+                'queue': que_2, 'loop': self.test_cfg['loop_count'], 'created_bucket': bucket_list}
+        proc_write_op = multiprocessing.Process(target=self.dtm_obj.perform_write_op, kwargs=args)
+        proc_write_op.start()
+
+        time.sleep(self.delay)
```

Comment:
add a logger whenever we are sleeping for some time so test doesn't look like stuck